libafl/src/stages/mod.rs

```diff
@@ -88,6 +88,8 @@ where
     /// On restart, this will be called again.
     /// As long as [`Stage::clear_restart_progress`], all subsequent calls happen on restart.
     /// Returns `true`, if the stage's [`Stage::perform`] method should run, else `false`.
+    /// Who should set this to true?
+    /// In general, if your stage does not involve random behavior, or it is deterministic, then you should always set this to false
```

Comment:
Why?

Reply:
why not. 
if your stage is deterministic then it will just keep crashing and restarting again and again.

Reply:
It's wrong, `should run` is `true` if the stage should run.
So it needs to be true always, unless after a restart that failed.